### ISSUE TITLE
Commands should have better /help messages

### ISSUE THREAD

**Issue by willkroboth** [JorelAli/CommandAPI#215]
**Problem**
At the moment, when you register a command with CommandAPI, using /help on the command isn't very helpful. Since the commands are being registered as Minecraft commands, the help page looks something like /stop's:
```
---------- Help: /stop ----------
Description: A Mojang provided command.
Usage: stop
```
Another example for a command registered with CommmandAPI:
```
---------- Help: /tphere ----------
Description: A Mojang provided command.
Usage: tphere
```
**Suggestion**
It would be much better if these help messages could be modified to look more like those generated for Bukkit commands. For example, /help's:
```
---------- Help: /help ----------
Description: Shows the help menu
Usage: /help <pageNumber>
/help <topic>
/help <topic> <pageNumber>
Aliases: ?
```
An example of what the previous CommandAPI command could look like:
```
---------- Help: /tphere ----------
Description: Teleports the selected entities to the executor
Usage: /tphere <entities>
```
**Possible Solution**
To allow server admins to modify and add help topics, Spigot and Paper servers have a help.yml file. For example, if you put the following into the help.yml file, using '/help tphere' will show what I suggested.
```
amended-topics:
  /tphere:
      fullText: "&6Description: &FTeleports the selected entities to the executor\n&6Usage: &F/tphere <entities>"
```
Something like the following code that I made for one of my plugins could be used to automatically generate and put this information into the help.yml file:
```
// code for generating and inserting the help entry
// not all the code necessary to make this work, but the main idea is here
private void setup_help(String name, String description, String arguments, List<String> aliases, FileConfiguration helpYML){
        // see https://bukkit.fandom.com/wiki/Help.yml for formatting and color
        logger.info("Generating help entry for " + name + " in help.yml");

        ConfigurationSection amendedTopics = helpYML.getConfigurationSection("amended-topics");
        if (amendedTopics == null){
            if (debugMode) logger.info("amended-topics not found, so it is being created.");
            amendedTopics = helpYML.createSection("amended-topics");
        }

        if (amendedTopics.isConfigurationSection("/" + name)){
            if (debugMode) logger.info("/" + name + " already has information. This information will be moved to old/" + name);
            amendedTopics.set("old/" + name, amendedTopics.getConfigurationSection("/" + name));
            amendedTopics.set("/" + name, null);
        }
        ConfigurationSection commandHelp = amendedTopics.createSection("/" + name);

        String aliasesString = "";
        if (aliases.size() != 0) {
            aliasesString = aliases.toString();
            aliasesString = aliasesString.substring(1, aliasesString.length() - 1);
            aliasesString = "&6Aliases: &F" + aliasesString;
        }
        if (debugMode) logger.info("aliasesString is " + aliasesString);

        String fullText =
                "&6Description: &F" + description + "\n" +
                "&6Usage: &F/" + name + " " + arguments + "\n" +
                aliasesString;
        commandHelp.set("fullText", fullText);
        if (debugMode) logger.info("fullText set to:\n" + fullText);

        if (debugMode) logger.info("Setting up help for " + aliases.size() + " aliases.");
        fullText = "&EAlias for /" + name + "\n" + fullText;
        if (debugMode) logger.info("fullText updated to:\n" + fullText);

        logger.increaseIndentation();
        for (String alias:aliases){
            if (debugMode) logger.info("Setting help for alias: " + alias);

            if (amendedTopics.isConfigurationSection("/" + alias)){
                if (debugMode) logger.info("/" + alias + " already has information. This information will be moved to old/" + alias);
                amendedTopics.set("old/" + alias, amendedTopics.getConfigurationSection("/" + alias));
                amendedTopics.set("/" + alias, null);
            }
            ConfigurationSection aliasHelp = amendedTopics.createSection("/" + alias);
            aliasHelp.set("fullText", fullText);
        }
        logger.decreaseIndentation();
}
```
**Downsides**
- This method may override information the user has already put into the help.yml file. I try to avoid this by moving old information to 'old/commandname' but that just moves the problem to another spot.
- As far as I know, there is no way to modify the help.yml file before the server loads it, which means that the server must be restarted before the changes have an effect.

There may be a better solution that avoids these problems, but I think that something like this would be a good addition to the CommandAPI.

**Comment by JorelAli** [comment]
See also issue #157. I am well aware that help topics suck and the process of trying to figure out how to register helpful help topics is not easy! I'll review your suggestions above at some point, they look pretty in-depth and helpful.

One of the main difficulties is that we're literally registering vanilla commands and trying to wrangle with Bukkit to change the help for specific vanilla commands is difficult.

**Comment by JorelAli** [comment]
I've managed to have some good results with help topic implementation!
I'll schedule a time implementation of this for 6.1.0.

**Comment by JorelAli** [comment]
Implemented in 6.4.0.
